addon/-private/system/identity-map.js

```diff
@@ -0,0 +1,28 @@
+import RecordMap from './record-map';
+import EmptyObject from './empty-object';
+
+export default class IdentityMap {
+  constructor() {
+    this._map = new EmptyObject();
```

Comment:
`Object.create(null)` may be fine, here. EmptyObject was more of a trick for mega mega hotspots to work-around a strange perf issue in older v8's.

In this case, I believe `Object.create(null)` will be best. (less caveats etc)